pkgs/os-specific/linux/kernel/generic.nix

```diff
@@ -108,6 +108,7 @@ let
 
   commonStructuredConfig = import ./common-config.nix {
     inherit lib stdenv version;
+    rustAvailable = lib.meta.availableOn stdenv.hostPlatform rustc;
```

Comment:
I don't think this is quite right — even if rustc is not available for the platform we're building for, rustc on the build platform might still be able to build Linux for it. I added a mechanism recently for doing this check properly — see gstreamer for an example.

Reply:
Ehh apologies, you even mentioned gstreamer part before already! How embarrassing :see_no_evil: 
Will take a look.

Reply:
We shouldn't — I was doing it to 64-bit MIPS recently…